config/install/field.storage.user.field_age_group.yml

```diff
@@ -0,0 +1,35 @@
+langcode: en
+status: true
+dependencies:
+  module:
+    - options
+    - user
+id: user.field_age_group
+field_name: field_age_group
+entity_type: user
+type: list_string
+settings:
+  allowed_values:
```

Comment:
1) I'm not sure about current values, they seem hard to work with. Wouldn't it be better to save one integer value of the max value?
For example, 19 for 15-19, 30 for 20-30.

2) Right now there are some overlaps. For example, if user is 30 years old, then he belongs to both groups: 20-30 and 30-40. I believe it should be changed, but let's clarify with Zohar first.